### PR TITLE
fix(server/cookies): Making the cookie better

### LINKED ISSUE #1228: [BUG] Use secure cookie
If the login is done over a secure HTTPS connection, the login cookie should have the `Secure` attribute set.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies#restrict_access_to_cookies

### PR DESCRIPTION
**Pull request by samip5**
Cookie should have SameSite=Stict and Secure if served via https, otherwise just SameSite=Strict set.

I'm not sure as to how I should fix the tests as I need the request params for it, which probably needs to be done in a mock of some sort?

Fixes #1228.

### CODE REVIEW

File: server/apps/immich/src/modules/immich-jwt/immich-jwt.service.ts [hunk 1]
```diff
@@ -22,11 +23,19 @@ export class ImmichJwtService {
...
     const maxAge = 7 * 24 * 3600; // 7 days
 
-    const accessTokenCookie = `${IMMICH_ACCESS_COOKIE}=${loginResponse.accessToken}; HttpOnly; Path=/; Max-Age=${maxAge}`;
-    const authTypeCookie = `${IMMICH_AUTH_TYPE_COOKIE}=${authType}; Path=/; Max-Age=${maxAge}`;
+    let accessTokenCookie = '';
+    let authTypeCookie = '';
+
+    if (request.secure) {
+      accessTokenCookie = `${IMMICH_ACCESS_COOKIE}=${loginResponse.accessToken}; Secure; Path=/; Max-Age=${maxAge}; SameSite=Strict;`;
+      authTypeCookie = `${IMMICH_AUTH_TYPE_COOKIE}=${authType}; Path=/; Max-Age=${maxAge}; SameSite=Strict;`;
```
Comment (jrasm91) [inline server/apps/immich/src/modules/immich-jwt/immich-jwt.service.ts:34]: Is this one missing secure?

**Review by jrasm91 — COMMENTED** [review]
It might be easier to pass req.secure instead.

File: server/apps/immich/src/modules/immich-jwt/immich-jwt.service.ts (hunk 1 above)
Comment (samip5) [reply server/apps/immich/src/modules/immich-jwt/immich-jwt.service.ts:34]: Oh, yes it is. Good catch.

**Comment by samip5** [comment]
> It might be easier to pass req.secure instead.

Could you elaborate as to what do you mean by that exactly?

**Comment by jrasm91** [comment]
Sending `req.secure` instead of `req` into the service might make testing easier.

**Comment by samip5** [comment]
> Sending `req.secure` instead of `req` into the service might make testing easier.

Indeed it does, and now I changed it to do that instead which made fixing the tests way easier. :)

**Pull request merged**